packages/gatsby/package.json

```diff
@@ -115,8 +115,8 @@
     "raw-loader": "^0.5.1",
     "react-dev-utils": "^4.2.3",
     "react-error-overlay": "^3.0.0",
-    "react-hot-loader": "^4.12.18",
-    "redux": "^4.0.4",
+    "react-hot-loader": "^4.12.19",
+    "redux": "4.0.4",
```

Comment:
I pinned `redux` because updating it was breaking our typings for some reason and failed `yarn typecheck` as a result

@blainekasten Could you take a look at this with the latest `redux` and see what's up?

Reply:
@sidharthachatterjee I'm guessing this https://github.com/reduxjs/redux/pull/3679

I didn't see what `yarn typecheck` failed with.

Reply:
```
packages/gatsby/src/redux/index.ts:59:5 - error TS2345: Argument of type 'IReduxState' is not assignable to parameter of type '{ [x: string]: unknown; }'.
  Index signature is missing in type 'IReduxState'.

59     initialState,
       ~~~~~~~~~~~~
```

Reply:
Okay fixed and pushed. This is a great example of the downsides of TS. This took me 30m to figure out, even though i'm pretty competent in TS. And the fix does not feel obvious or should be required.. so that's fun :\ 

Ultimately I had to look at the commits that landed in the new release and found this PR https://github.com/reduxjs/redux/pull/3485/files

reading the comments made it sort of obvious. But most libs won't go through this effort of documenting their code.